core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2582,13 +2582,21 @@ public void resetFile(long fileId)
 
   @Override
   public void setAttribute(AlluxioURI path, SetAttributeOptions options)
-      throws FileDoesNotExistException, AccessControlException, InvalidPathException {
+      throws FileDoesNotExistException, AccessControlException, InvalidPathException,
+      IOException {
     Metrics.SET_ATTRIBUTE_OPS.inc();
     // for chown
     boolean rootRequired = options.getOwner() != null;
     // for chgrp, chmod
     boolean ownerRequired =
         (options.getGroup() != null) || (options.getMode() != Constants.INVALID_MODE);
+    if (options.getOwner() != null && options.getGroup() != null) {
+      try {
+        checkUserBelongsToGroup(options.getOwner(), options.getGroup());
+      } catch (IOException e) {
+        throw new IOException("Could not setOwner for " + path.toString() + ".", e);
```

Comment:
Since checkUserBelongToGroup method is kind of a utility method, the error message contained in the exception thrown by checkUserBelongToGroup is not specific and does not tell the end user which operation caused the exception. Therefore, to provide user with more specific information, we need to add some information to its message, but propagate the exception. Is this correct?